geps/gep-1494/index.md

```diff
@@ -0,0 +1,144 @@
+# GEP-1494: HTTP Auth in Gateway API
+
+* Issue: [#1494](https://github.com/kubernetes-sigs/gateway-api/issues/1494)
+* Status: Provisional
+
+(See status definitions [here](/geps/overview/#gep-states).)
+
+
+## TLDR
+
+Provide a method for configuring **Gateway API implementations** to add HTTP Authentication for north-south traffic. The method may also include Authorization config if practical. At the time of writing, this authentication is only for ingress traffic to the cluster.
+
+
+## Goals
+
+(Using the [Gateway API Personas](https://gateway-api.sigs.k8s.io/concepts/roles-and-personas/))
+
+* A way to configure a Gateway Implementation to perform Authentication (at least), with optional Authorization on behalf of Ana the Application Developer.
+
+* A way for Chihiro the Cluster Admin to configure a default Authentication and/or Authorization config for some set of HTTPRoutes.
+
+* Optionally, a way for Ana to have the ability to disable Authentication and/or Authorization for specific routes when needed, allowing certain routes to not be protected.
+
+
+## Non-Goals
+
+* Handling all possible authentication and authorization schemes. Handling a (preferably large) subset of authentication and authorization is acceptable.
+
+
+## Deferred Goals
+
+* (Not performed during the Provisional Phase) Defining the API or configuration to be used.
+
+## Introduction
+
+A common request for Gateway API has been a way to direct implementations to automatically request various forms of authentication (AuthN) and authorization (AuthZ). (see the [GEP Issue](https://github.com/kubernetes-sigs/gateway-api/issues/1494) for some of the history.)
+
+**Authentication (AuthN for short)** refers to proving the requesting party's identity in some fashion.
+
+**Authorization (AuthZ for short)** refers to making decisions about what a party can access _based on their Authenticated identity_.
+
+In this document, we'll use Auth* to represent "AuthN and maybe AuthZ as well".
+
+This capability is useful for both Ana (who wants to have something for AuthN and AuthZ that ensures security without it needing to be built into her app), and Chihiro (who wants to be able to ensure that the platform as a whole conforms to some given level of security).
+
+### Common authentication methods
+
+Before discussing any proposed Auth* solution, it's important to discuss some AuthN methods that are often used in securing modern applications.
+
+#### Basic HTTP Auth
+
+In Basic HTTP Auth, a server asks for authentication as part of returning a `401` status response, and the client includes an `Authorization` header that includes a Base64-encoded username and password.
+
+Because the password is only _encoded_ and not _encrypted_, Basic Auth is totally unsafe when used outside of an encrypted session (like a HTTPS connection).
```

Comment:
While the "raw" passwords in basic auth have additional issues (long lived, impersonating human on login pages), JWT and afaik OAUTH/OIDC are not generally not secure against replay attacks.

I.e. besides mTLS (which enforces confidentiality) all mechanisms are insecure in plaintext messages.

I think it's best to move the notion of requiring encryption for safe usage out of Basic Auth and potentially add a note here that basic auth has an additional issue with long-lived (potentially higher power) tokens being exchanged.